src/dataflow/analysis/tests/flow-graph-tests.ts

```diff
@@ -216,6 +216,41 @@ describe('FlowGraph validation', () => {
     assert.sameMembers(result.failures, [`'check bar is trusted' failed for path: P1.foo -> P2.bar`]);
   });
 
+  it('fails when a "not tag" is claimed and the tag is checked for', async () => {
```

Comment:
Other important test cases:
1. check that it deletes tags correctly: i.e. the path `is tag -> is not tag -> check is tag` fails
2. check that tags can get reintroduced. i.e. the path `is tag -> is not tag -> is tag -> check is tag` succeeds

Reply:
Thanks. I meant to write more tests but forgot. Done.